common/lib/common-definitions/src/logger.ts

```diff
@@ -59,6 +59,29 @@ export interface ITelemetryPerformanceEvent extends ITelemetryGenericEvent {
     duration?: number; // Duration of event (optional)
 }
 
+/**
+ * A property to be logged to telemetry containing both the value and the tag
+ */
+export interface ITaggedTelemetryPropertyType {
+    value: TelemetryEventPropertyType,
+    tag: string
+}
+
+/**
+ * Property bag containing a mix of value literals and wrapped values along with a tag
+ */
+export interface ITaggableTelemetryProperties {
```

Comment:
Ok the last thing I'm wondering is if we should just tackle #5560 here and now, and just update `ITelemetryProperties` to optionally have tagged props. (i.e. replace `ITelemetryProperties` with this interface but keep the name).

We should also add an optional flag `supportsTags?: boolean` to `ITelemetryBaseLogger` so we can be sure to hand back only untagged properties if true, for back compat.

@vladsud this is what I want to discuss with you :)

Reply:
And add something in BREAKING.MD, although due to `supportsTags`, taking no action will be ok in order to merge in the new package versions for hosts.  But hopefully they would update ASAP anyway.

Reply:
Actually I think supportsTags should be required, and the BREAKING.MD comment can say just set it to false at first?

Reply:
I don't see how that works with our versioning scheme that support mixing between the layers, or how clients use that field. this seems like something that needs to deploy dark, and then light up after fully propagated

Reply:
Oh... I guess it does need to be optional.  Let me see if I can describe how this will go.

So the host with loader integrated will provide the `ITelemetryBaseLogger` impl, which at first will remain without `supportsTags` (because it's old). All loggers built on that should pass through that value, so even in a newer runtime, `supportsTags` will be `undefined`. So the code in `telemetry-utils` will continue to handle tags as-is (because it will do so unless `supportsTags === true`).

Then one day OWH will update their base logger to support tags and will set that flag to `true`, and then it will light up.

Then in the distant future we can update this flag to be required, and then required with type `true` instead of `boolean`, and then get rid of it altogether.

Reply:
that sounds right, also, the typing should be `supportsTags?: true` no one should ever set it to `false`

Reply:
@anthony-murphy can we update BREAKING.MD even though it's not strictly a breaking change?  I'd like to draw attention to it.

Reply:
Added